Framework/Core/src/DeviceSpecHelpers.cxx

```diff
@@ -744,6 +744,34 @@ void DeviceSpecHelpers::dataProcessorSpecs2DeviceSpecs(WorkflowSpec const& workf
   }
 }
 
+void DeviceSpecHelpers::reworkShmSegmentSize(std::vector<DataProcessorInfo>& infos)
+{
+  int64_t segmentSize = 0;
+  for (auto& info : infos) {
+    auto it = std::find(info.cmdLineArgs.begin(), info.cmdLineArgs.end(), "--shm-segment-size");
+    if (it == info.cmdLineArgs.end()) {
+      continue;
+    }
+    auto value = it + 1;
+    if (value == info.cmdLineArgs.end()) {
+      throw std::runtime_error("--shm-segment-size requires an argument");
+    }
+    char* err = nullptr;
+    int64_t size = strtoll(value->c_str(), &err, 10);
+    if (size > segmentSize) {
+      segmentSize = size;
+    }
+    info.cmdLineArgs.erase(it, it + 2);
+  }
+  if (segmentSize == 0) {
+    segmentSize = 2000000000LL;
```

Comment:
We'll eventually want to back the shm segments with huge pages, where the size become important (multiple of page size). Perhaps this should be (2ULL << 30)

Reply:
Do we want GB huge pages or 2/4 MB? GB huge pages need to be allocated at boot time if I remember correctly?

Reply:
I think 2M would be fine for us. Don't think there would be much benefit going to 1G. There are several reservation methods with different flexibility: kernel boot parameter, early during boot, etc. We'll have to see what works best for us.

Reply:
Notice that that's the current default in FairMQ, but I agree a 1ULL <<31 is probably better in any case. This could should actually be handled at FairMQ level, IMHO: always snap and align to page boundaries...